nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf

```diff
@@ -46,45 +45,40 @@ def helpMessage() {
     """.stripIndent()
 }
 
-/*
- * SET UP CONFIGURATION VARIABLES
- */
-
-// Show help emssage
-if (params.help){
+// Show help message
+if (params.help) {
     helpMessage()
     exit 0
 }
 
+/*
+ * SET UP CONFIGURATION VARIABLES
+ */
+
 // Check if genome exists in the config file
 if (params.genomes && params.genome && !params.genomes.containsKey(params.genome)) {
     exit 1, "The provided genome '${params.genome}' is not available in the iGenomes file. Currently the available genomes are ${params.genomes.keySet().join(", ")}"
 }
 
 // TODO nf-core: Add any reference files that are needed
 // Configurable reference genomes
-fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
-if ( params.fasta ){
-    fasta = file(params.fasta)
-    if( !fasta.exists() ) exit 1, "Fasta file not found: ${params.fasta}"
-}
 //
 // NOTE - THIS IS NOT USED IN THIS PIPELINE, EXAMPLE ONLY
-// If you want to use the above in a process, define the following:
+// If you want to use the channel below in a process, define the following:
 //   input:
-//   file fasta from fasta
+//   file fasta from ch_fasta
 //
-
+params.fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
+if (params.fasta) { ch_fasta = file(params.fasta, checkIfExists: true) }
```

Comment:
It's better to avoid using `file` objects, as this has caused issues on AWS and other systems. This isn't your bug, it was already in the template, but we should use a channel instead..

Also, `checkIfExists` is specifically for [`Channel.fromPath`](https://www.nextflow.io/docs/latest/channel.html#frompath), so I'd be kind of surprised if it worked on a `file` object like this..


```suggestion
if (params.fasta) {
  ch_fasta = Channel.fromPath(params.fasta, checkIfExists: true)
            .ifEmpty { exit 1, "Fasta file not found: ${params.fasta}" }
}
```

Reply:
Ok, so @pditommaso says _"you should always use `file()`"_ and it turns out that `checkIfExists` _does_ work on `file()` as well as `Channel.fromPath()`.

So basically, everything I said here was wrong and what yo had already written was perfect 😆